### ISSUE TITLE
Bug - Ships built with Space Dock get incorrect armor values

### ISSUE THREAD

**Issue by drDragonSmoke** [tetracorp/k240#15]
This is an odd one. Ships built with the Space Dock get an armor value slightly lower than what is expected, and the difference depends on how many hardpoints it is built with. For example, the Destructor will end up with 37/40 armor when built with a hardpoint in all 3 of its hardpoint slots, and 34/40 armor when built with no hardpoints. I have yet to find the part of the code that affects this, but the math seems to be:
6 ÷ hardpoint slots = armor subtracted per unfilled hardpoint slot
6 ÷ hardpoint slots - 1 = armor subtracted per filled hardpoint slot

In short, all ships that can be built with the Space Dock end up with 6 armor points less than what is expected when built with no hardpoints. The Fleet Battle Ship is the only ship that ends up with the expected armor value, but only when built with all 6 of its hardpoint slots filled.

**Comment by tetracorp** [comment]
Well spotted. It's amazing that we're stlil finding bugs in this game.

I think I've found the relevant code. At `loop1463A`, it iterates over all six hardpoint slots and adds the relevant value from `arrHPArmrValues`. However, any of the six slots without a hardpoint in it is represented by `0xff`, or `-1`. This is the same whether the slot is empty, or the slot is absent entirely because the ship has fewer than six hardpoints.

The OSD code therefore adds an offset of `-1` to `arrHPArmrValues` and retrieves the two-byte value before, which happens to be `0xffff`, also -1. Therefore, each empty or absent slot costs the ship -1 Armour.

If you look at `loop1148A`, which is the equivalent code for the Construction Yard, it has a `BMI` check specifically to prevent a -1 from being added to Armour. The check is missing from the OSD version of the code.

**Comment by drDragonSmoke** [comment]
Oh, brilliant! Thank you. Bug fix:
```
loop1463A:
	MOVE.B	(A3)+,D6		;1463a: 1c1b
	BMI.S	_skipArmBug		;;;
	EXT.W	D6			;1463c: 4886
	ADD.W	D6,D6			;1463e: dc46
	ADD.W	(0,A4,D6.W),D1		;14640: d2746000
_skipArmBug:				;;;
```
